airbyte-integrations/connectors/source-github/source_github/source.py

```diff
@@ -88,6 +88,43 @@ def _get_authenticator(token: str):
         tokens = [t.strip() for t in token.split(TOKEN_SEPARATOR)]
         return MultipleTokenAuthenticator(tokens=tokens, auth_method="token")
 
+    @staticmethod
+    def _get_branches_data(selected_branches: str, full_refresh_args: Dict[str, Any] = None) -> Tuple[Dict[str, str], Dict[str, List[str]]]:
+        selected_branches = set(filter(None, selected_branches.split(" ")))
+
+        # Get the default branch for each repository
+        default_branches = {}
+        repository_stats_stream = RepositoryStats(**full_refresh_args)
+        for stream_slice in repository_stats_stream.stream_slices(sync_mode=SyncMode.full_refresh):
+            default_branches.update(
+                {
+                    repo_stats["full_name"]: repo_stats["default_branch"]
+                    for repo_stats in repository_stats_stream.read_records(sync_mode=SyncMode.full_refresh, stream_slice=stream_slice)
```

Comment:
sorry can you explain what this for loop is doing?

Reply:
We get the default branch for each of the repositories, so that if nothing is specified in the `branches` parameter in `spec.json`, then use it.